concrete/src/Permission/PermissionServiceProvider.php

```diff
@@ -2,13 +2,15 @@
 namespace Concrete\Core\Permission;
 
 use Concrete\Core\Foundation\Service\Provider as ServiceProvider;
-use Concrete\Core\Page\Type\Validator\Manager as ValidatorManager;
-use Concrete\Core\Page\Type\Saver\Manager as SaverManager;
 
 class PermissionServiceProvider extends ServiceProvider
 {
     public function register()
     {
+        $app = $this->app;
         $this->app->singleton('Concrete\Core\Permission\Inheritance\Registry\BlockRegistry');
+        $this->app->singleton('permission/access/entity/factory', function() use ($app) {
```

Comment:
you dont need the `$app = $this->app`
Just change this line to
`$this->app->singleton('permission/access/entity/factory', function($app) {`